x/wasm/internal/keeper/handler_plugin.go

```diff
@@ -19,32 +23,46 @@ type MessageHandler struct {
 }
 
 func NewMessageHandler(router sdk.Router, customEncoders *MessageEncoders) MessageHandler {
-	encoders := DefaultEncoders().Merge(customEncoders)
+	encoders := DefaultEncoders(nil, nil).Merge(customEncoders)
 	return MessageHandler{
 		router:   router,
 		encoders: encoders,
 	}
 }
 
-type BankEncoder func(sender sdk.AccAddress, msg *wasmTypes.BankMsg) ([]sdk.Msg, error)
+func NewMessageHandlerV2(router sdk.Router, channelKeeper types.ChannelKeeper, capabilityKeeper types.CapabilityKeeper, customEncoders *MessageEncoders) MessageHandler {
+	encoders := DefaultEncoders(channelKeeper, capabilityKeeper).Merge(customEncoders)
+	return MessageHandler{
+		router:   router,
+		encoders: encoders,
+	}
+}
+
+type BankEncoder func(sender sdk.AccAddress, msg *cosmwasmv1.BankMsg) ([]sdk.Msg, error)
 type CustomEncoder func(sender sdk.AccAddress, msg json.RawMessage) ([]sdk.Msg, error)
-type StakingEncoder func(sender sdk.AccAddress, msg *wasmTypes.StakingMsg) ([]sdk.Msg, error)
-type WasmEncoder func(sender sdk.AccAddress, msg *wasmTypes.WasmMsg) ([]sdk.Msg, error)
+type StakingEncoder func(sender sdk.AccAddress, msg *cosmwasmv1.StakingMsg) ([]sdk.Msg, error)
+type WasmEncoder func(sender sdk.AccAddress, msg *cosmwasmv1.WasmMsg) ([]sdk.Msg, error)
+type IBCEncoder func(ctx sdk.Context, sender sdk.AccAddress, source cosmwasmv2.IBCEndpoint, msg *cosmwasmv2.IBCMsg) ([]sdk.Msg, error)
```

Comment:
This is all the extra info we need? Not an external keeper?

Nice to make this simple as possible

Reply:
`ChannelKeeper` and `CapabilityKeeper` were passed to the constructor.